[FancyZones] Scrollable Editor

davidegiacometti opened the pull request:
## Summary of the Pull Request

Added scrolling to Fancy Zones Editor

## PR Checklist
* [x] Applies to #738
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA
* [x] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #738

## Info on Pull Request

Tested on:

- 1920x1080
- 1024x768 `SizeToContent="Height"` no more needed

![FZ1](https://user-images.githubusercontent.com/25966642/99452821-5796e380-2924-11eb-89aa-5acc4ee84f55.gif)
![FZ2](https://user-images.githubusercontent.com/25966642/99452825-5960a700-2924-11eb-92d0-401bd634b447.gif)


## Validation Steps Performed

Test FancyZones editor with scrolling

niels9001 commented:
#6899 addresses this as well. But this PR would be an easier merge.

davidegiacometti commented:
> #6899 addresses this as well. But this PR would be an easier merge.

Didn't know about FancyZones UI/UX review 😅

htcfreek commented:
Merging this could conflict with #6562.

enricogior commented:
Hi @davidegiacometti 
thank you very much for your contribution.

We should wait before merging this PR for a couple of reasons.
We don't want to revert previous changes for the editor UI for small screens, it's OK to add scrolling but it should still look like this
![image](https://user-images.githubusercontent.com/3206696/97151467-d490cc00-176f-11eb-8d81-3d002f4810de.png)
and not like this since it present a discoverability issue
![image](https://user-images.githubusercontent.com/3206696/97151897-71536980-1770-11eb-9e0f-f76b7150d66b.png)

We are also in the process of redesign the UI https://github.com/microsoft/PowerToys/issues/1032
Changes to the current UI will require a new accessibility assessment, so instead of doing it twice (for your changes and then again for the new UI) let's work together to update/improve the UI after https://github.com/microsoft/PowerToys/pull/6562 is merged (that also introduces changes to the UI).

I'm going to convert this PR to draft for now, OK?

enricogior commented:
@davidegiacometti 
we just merged the editor multi-monitor support.
The UI is different when multiple monitors are present, so make sure that adding the scroll bar doesn't change how the UI is rendered in that case.
If you don't have a multi monitor configuration, you can simulate it passing arguments in debug mode to the editor.
In case let me know if you need any hint on how to do it.

davidegiacometti commented:
@enricogior I have fixed scrolling for both sizes but I think the UI can be improved further.
I have updated the first post with the result.
Also the Close button appear only when there is more than one monitor. Not sure if a bug or a feature 😄

enricogior commented:
@davidegiacometti 

> I think the UI can be improved further

This is the next step for the UI: https://github.com/microsoft/PowerToys/pull/6899

> Also the Close button appear only when there is more than one monitor. Not sure if a bug or a feature 😄

That is by design, because the single monitor UI has not been changed, and applying a layout closes the editor.
On multi-monitor, applying a layout doesn't close the editor, so a dedicated Close button is needed.

enricogior commented:
@davidegiacometti 
we need to revisit these changes because I don't think this is an improved UI:

![image](https://user-images.githubusercontent.com/3206696/99503624-b8eba080-297e-11eb-9e83-39a9c9e46cbe.png)

It doesn't have a visible scroll bar and for a first time user it's not immediately clear there are two more templates (Grid and Priority Grid).
On large screens is not very useful to have a fixed window size, it's not a good usage of the screen estate, we should adjust the windows size based on the space available.
Given we will not accept more changes for 0.27 after this Friday, it might be better to work with @niels9001 and join forces to add scroll bar to his upcoming PR for the updated UI for the December release.

davidegiacometti commented:
@enricogior now the scroll bars are visibles.
I agree that the fixed size is not optimal on large screens, than feel free to close the PR without merging and release this in December with the new UI.
![image](https://user-images.githubusercontent.com/25966642/99577087-1e6d7a80-29db-11eb-96aa-81bd5b6df15f.png)

enricogior commented:
@davidegiacometti 
thanks for fixing the scrollbars visibility.
I've changed the PR to draft since now it's too late to makes other changes and test them for 0.27.
Niels is starting to work on the revised UI, so we can wait for his work and then decide what can be reused from this PR, in case we will close it and open a new one.

davidegiacometti commented:
@enricogior I think this can be closed since @niels9001 is working on https://github.com/microsoft/PowerToys/pull/8148

enricogior commented:
@davidegiacometti 
OK. Thanks.
